src/modules/project/funding/hooks/useRefferalWithProjectName.ts

```diff
@@ -0,0 +1,15 @@
+import { useAtomValue } from 'jotai'
+
+import { refferalAtom } from '../../pages/projectView/state/affiliateAtom'
+
+export const useRefferalWithProjectName = (projectName?: string) => {
+  const refferals = useAtomValue(refferalAtom)
+
+  if (!projectName) return undefined
+
+  const currentRefferal = refferals.find((r) => r.projectName === projectName)
```

Comment:
I suggest we rename all "referral" to "affiliate". We call everything affiliate in the backend too and it would help us avoid confusion if we add a separate referral program in the future.